### DISCUSSION TITLE
Export `setRemixDevLoadContext()` from `@remix-run/dev`

### DISCUSSION THREAD

**Discussion by zirkelc** [remix-run/remix#9742] [Proposals]
I would like to suggest to export the `setRemixDevLoadContext()` from the `@remix-run/dev` package so it can be used to set the load context during development.

The `setRemixDevLoadContext()` is exported here:
https://github.com/remix-run/remix/blob/bbead33bf530be030d16a1466b52cd99d077382a/packages/remix-dev/vite/plugin.ts#L575-L579

But the `vite/index.ts` doesn't re-export it: 
https://github.com/remix-run/remix/blob/bbead33bf530be030d16a1466b52cd99d077382a/packages/remix-dev/vite/index.ts#L4-L18

**Comment by michaelboyles** [comment]
I want this feature too, but I don't think just exporting that thing is the right approach just because it happens to already exist.

When that was introduced (https://github.com/remix-run/remix/commit/44bac2982cc7daeab5624a332d1d8f3b186c5707), the cloudflare preset was calling it. Now nothing calls it. The cloudflare support was changed from a preset to a separate vite plugin.

I'd much rather write

```js
plugins: [
    remix({
        presets: [devContextLoader(whatever)]
    })
]
```

or

```js
remix({
     devContextLoader: whatever
})
```

than what exporting the function would give you, which is two things which are coupled together, with almost no indication that that's the case

```js
setRemixDevLoadContext(whatever);

// ...

plugins: [remix()]
```

**Reply by zirkelc** [reply]
I fully agree with your point! If someone from the Remix team would agree to any of the suggested options, I'd be happy submit a PR.
